### ISSUE TITLE
Strangeness with micromamba executable

### ISSUE THREAD

**Issue by abalter** [mamba-org/mamba#2331]
### Troubleshooting docs

- [X] My problem is not solved in the Troubleshooting docs

### Search tried in issue tracker

executable

### Latest version of Mamba

- [X] My problem is not solved with the latest version

### Tried in Conda?

Not applicable

### Describe your issue

I'm trying to find a way to make a symlink called `conda` that points to the micromamba executable. I can see that `micromamba` is in my path. I created a symlink called `conda` in that same directory that points to `micromamba`. When I type `conda -h` I get the help  file. However, the commands `conda activate` and `conda deactiate` don't work.

I clearly don't understand how the executable works. Is there a way to do what I want to do?

Adding `alias conda="micromamba"` to my .bashrc does work as I wish. But I'd like something more robust.

The output below is after doing a clean uninstall/reinstall of micromamba and then opening a new shell:

```
balter@spectre:~$ which conda
balter@spectre:~$ which micromamba
/home/balter/.local/bin/micromamba
balter@spectre:~$ micromamba actiate
The following argument was not expected: actiate
Run with --help for more information.
balter@spectre:~$ micromamba activate
(base) balter@spectre:~$ micromamba deactivate
balter@spectre:~$ echo $PATH | grep -n micromamba
1:/home/balter/.local/bin:/home/balter/micromamba/condabin:~/expand_and_flatten_vcf:/home/balter/mamba/bin:/home/balter/utilities/amd_templates/setup:/home/balter/utilities/project-utilities:/home/balter/utilities/analysis-utilities:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0/:/mnt/c/WINDOWS/System32/OpenSSH/:/mnt/c/Program Files (x86)/NoteBook FanControl/:/mnt/c/Program Files/Meld/:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/ariel/miniconda3:/mnt/c/Users/ariel/miniconda3/Library/mingw-w64/bin:/mnt/c/Users/ariel/miniconda3/Library/usr/bin:/mnt/c/Users/ariel/miniconda3/Library/bin:/mnt/c/Users/ariel/miniconda3/Scripts:/mnt/c/Users/ariel/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/ariel/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/ariel/AppData/Local/Google/Cloud SDK/google-cloud-sdk/bin:/mnt/c/Users/ariel/AppData/Local/gitkraken/bin:/mnt/c/Users/ariel/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/ariel/AppData/Local/Pandoc/:/snap/bin
balter@spectre:~$ find /home/balter/micromamba -name condabin
balter@spectre:~$ #???????
balter@spectre:~$ ln -s /home/balter/.local/bin/micromamba /home/balter/.local/bin/conda
balter@spectre:~$ ls -l /home/balter/.local/bin
total 15564
lrwxrwxrwx 1 balter balter       34 Feb 28 11:59 conda -> /home/balter/.local/bin/micromamba
-rwxr-xr-x 1 balter balter 15935016 Feb 10 03:50 micromamba
balter@spectre:~$ which conda
/home/balter/.local/bin/conda
balter@spectre:~$ conda --version
__add_sys_prefix_to_path: command not found
Command '' not found, but can be installed with:
sudo apt install mailutils-mh  # version 1:3.14-1, or
sudo apt install mmh           # version 0.4-4
sudo apt install nmh           # version 1.7.1-11
sudo apt install termtris      # version 1.3-1ubuntu1
balter@spectre:~$ conda -h
__add_sys_prefix_to_path: command not found
Command '' not found, but can be installed with:
sudo apt install mailutils-mh  # version 1:3.14-1, or
sudo apt install mmh           # version 0.4-4
sudo apt install nmh           # version 1.7.1-11
sudo apt install termtris      # version 1.3-1ubuntu1
```

### mamba info / micromamba info

```shell
NA
```

### Logs

```shell
NA
```

### environment.yml

```yaml
NA
```

### ~/.condarc

```yaml
NA
```

**Comment by jonashaag** [comment]
Note that if you type `conda` in your shell you typically don’t invoke the executable directly but a shell function called `conda`. This might mess up some conceptions of how things behave, although I’m happy to discuss changes to the shell functions that make this less surprising.

**Comment by jonashaag** [comment]
The reason why we have a shell function is so that `activate` can change your current shell environment, which is not possible from a subprocess. Personally I think this design decision causes too much trouble and we should not offer `activate` in the first place.

**Comment by abalter** [comment]
**TL/DR:** I would like to create a symlink so I can use the command `conda` in place of `micromamba`. Is this possible?

So, as I was saying, `alias conda="micromamba"` after vanilla install works so long as the shell or process I launch sources my .bashrc file. In fact I tell me .profile to source my .bashrc, so I would never have this problem.

I'm trying to get my research group to try out conda and they are a little risk-averse. So I wrote a script to safely install in a way that can be completely removed. For instance, backs up their .bashrc file and then can replace it.

However, these folks are just learning linux and don't even know what a .bashrc file is. So, rather than use the alias method, I wanted to use a symlink in a folder that would get removed anyway during the uninstall. I tried a couple of micromamba subdirectories that are in my path, but that didn't work.

Do you have any suggestions?

**Comment by jonashaag** [comment]
If they use the shell initialization in bashrc from Conda then it’s unlikely that a symlink will work, for the reasons I stated above.

I don’t think you will have success teaching them Conda with something that is not a drop in replacement for Conda. My suggestion is to show them how to use Micromamba from the start.

**Comment by abalter** [comment]
That's what I'm doing. `micromamba` is just twice as long to type. And also they will see references to conda and instructions for using it all over the internet which could lead to confusion.

**Comment by abalter** [comment]
But, maybe it's better to just stick with micromamba and have them create their own alias if they want.

**Comment by abalter** [comment]
Hey, I've been using micromamba very successfully. I created the script I'm attaching to make it easy for others in my group. So far so good.

Can you tell me what the proper way to activate micromamba in a shell script is?

_setup_micromamba_

```
#!/bin/bash

action=
install_r_pkgs=
install_python_pkgs=

while getopts "ihurp" option;
do
    case $option in
        i)
            action="do_install"
        ;;

        r)
            install_r_pkgs="true"
        ;;

        p)
            install_python_pkgs="true"
        ;;

        h)
            action="show_help"
        ;;

        u)
            action="do_remove"
        ;;

        \?) # Invalid option
            echo "    ERROR: Invalid option '${option}'. Plese see setup_microconda -h"
            exit 1
        ;;
    esac
done

showHelp()
{
    cat<<EOF
    USAGE:
        install_miniconda [ -i ] [ -h ] [ -u ] [ -r ]

    DESCRIPTION:
        This script installs and configures the microconda package manager.

        This process makes no permanent changes to your computing environment.
        It creates a directory called micromamba in your home environment and
        adds a block of code to your .bashrc file.

        At any time, you can deactivate conda and return the the same linux
        environment you had previously. Alternatively, you can completely
        reverse this install and remove all components that were created or
        changed, including restoring your orignal .bashrc file.

        You can optionally install a basic set of the most commonly used r (-r)
        and python (-p) packages, which are listed below. If you don't use this
        option, be advised that your available software will be exactly the
        same as before you installed micromamba. You will need to install pacakges
        and activate your environment.

    OPTIONS:
        -i:  Install micromamba and a basic environment.
        -u:  Uninstall micromamba and remove all traces.
        -r:  Install a basic set of r packages. (see below)
        -p:  Install a basic set of python packages. (see below)
        -h:  Display this help message and quit.

    PACKAGES:
        These are the conda packages I chose to create a basic setup:
        R:
            - r-essentials: A metapackage of tons of common packages. See https://github.com/conda-forge/r-essentials-feedstock/blob/main/recipe/meta.yaml
            - r-tidyverse: A metapackage of packages in, yup, the tidyverse. See https://github.com/conda-forge/r-tidyverse-feedstock/blob/main/recipe/meta.yaml
            - r-rpostgresql
            - r-sqlite
            - r-tidytable (competition for dtplyr) https://markfairbanks.github.io/tidytable/reference/index.html
            - r-kableextra
        Python:
            - jupyterlab with nb_conda_kernels and jupytext
            - Pandas
            - Scipy
            - Matplotlib
            - Seaborn
            - SciKit-Learn
            - Beautiful Soup
EOF

}

doInstall()
{
    echo "doInstall"
    #exit

    rm -rf ~/micromamba

    if [[ ! -f ~/.bashrc ]]; then
        cat>~/.bashrc<<-EOF
        #!/bin/bash

        EOF
    else
        cp .bashrc before_miniconda_bashrc
    fi

    curl micro.mamba.pm/install.sh | bash

    source ~/micromamba/etc/profile.d/micromamba.sh
    source ~/.bashrc

    cat>~/.condarc<<-EOF
    channels:
    - conda-forge
    - bioconda
    EOF

    micromamba activate

    if [[ "$install_r_pkgs" == true ]]; then
        micromamba install r-essentials r-tidytable r-rpostgresql r-kableextra -y
    fi

    if [[ "$install_python_pkgs" == true ]]; then
        micromamba install jupyterlab pandas scipy matplotlib nb_conda_kernels jupytext seaborn scikit-learn beautifulsoup4 -y
    fi

    cat>>~/.bashrc<<-EOF
    alias conda="micromamba"
    conda activate

    EOF

}

doRemove()
{
    echo "Completely removing microconda"
    #exit

    rm -rf ~/miniconda
    rm -rf ~/.conda
    rm     ~/.condarc
    rm -r  ~/.local/bin/micromamba
    mv ~/before_miniconda_bashrc ~/.bashrc
}

confirmRemove()
{
    echo "confirmRemove"
    while true; do
        read -p "Do you wish to install this program? (y/n): " -n 1 yn
        case $yn in
            [Yy]* ) echo ""; doRemove; break;;
            [Nn]* ) echo ""; exit;;
            * ) echo "Please answer 'y' or 'n': ";;
        esac
    done
}


case ${action} in
    do_install)
        echo "do install"; doInstall
    ;;

    do_remove)
        confirmRemove
    ;;

    show_help)
        showHelp
    ;;
esac

exit 0
```

**Comment by jonashaag** [comment]
`micromamba activate myenv`

**Comment by abalter** [comment]
Hmmm. That wasn't working in my `.bashrc` file. I'll investigate further...
